fdbclient/common.go

```diff
@@ -45,6 +46,31 @@ const (
 	defaultTransactionTimeout = 5 * time.Second
 )
 
+func parseMachineReadableStatus(logger logr.Logger, contents []byte) (*fdbv1beta2.FoundationDBStatus, error) {
+	status := &fdbv1beta2.FoundationDBStatus{}
+	err := json.Unmarshal(contents, status)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(status.Client.Messages) > 0 {
+		logger.Info("found client message(s) in the machine-readable status", "messages", status.Client.Messages)
+	}
+
+	if len(status.Cluster.Messages) > 0 {
+		logger.Info("found cluster message(s) in the machine-readable status", "messages", status.Cluster.Messages)
+
+		// If the status is incomplete because of a timeout, return an error. This will force a new reconciliation.
+		for _, message := range status.Cluster.Messages {
+			if message.Name == "status_incomplete_timeout" {
+				return nil, fdbv1beta2.TimeoutError{Err: fmt.Errorf("found \"status_incomplete_timeout\" in cluster messages")}
+			}
+		}
+	}
+
+	return status, nil
```

Comment:
What about the case where the status doesn't list any processes? Are you assuming  that "messages" would be populated in that case?

Reply:
Makes sense to add this case too.